extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/remote_inference_client/text_translation.py

```diff
@@ -0,0 +1,291 @@
+    PromptMetadata,
+)
+from aiconfig.util.config_utils import get_api_key_from_environment
+from aiconfig.util.params import resolve_prompt
+
+
+# Circuluar Dependency Type Hints
+if TYPE_CHECKING:
+    from aiconfig.Config import AIConfigRuntime
+
+
+# Step 1: define Helpers
+def refine_completion_params(model_settings: dict[Any, Any]) -> dict[str, Any]:
+    """
+    Refines the completion params for the HF text translation api. Removes any unsupported params.
+    See https://github.com/huggingface/huggingface_hub/blob/main/src/huggingface_hub/inference/_client.py#L1703
+    for supported params. Note that client params DO NOT match API docs here:
+    https://huggingface.co/docs/api-inference/detailed_parameters#translation-task
+    """
+
+    supported_keys = {
+        "model",
+        "src_lang",
+        "tgt_lang",
+    }
+
+    completion_data = {}
+    for key in model_settings:
+        if key.lower() in supported_keys:
+            completion_data[key.lower()] = model_settings[key]
+
+    return completion_data
+
+
+def construct_output(response: str) -> Output:
+    metadata: dict[str, str] = {"raw_response": response}
+
+    output = ExecuteResult(
+        **{
+            "output_type": "execute_result",
+            "data": response,
+            "execution_count": 0,
+            "metadata": metadata,
+        }
+    )
+    return output
+
+
+class HuggingFaceTextTranslationParser(ParameterizedModelParser):
+    """
+    A model parser for HuggingFace text translation models.
+    """
+
+    def __init__(self, model_id: str = None, use_api_token=False):
+        """
+        Args:
+            model_id (str): The model ID of the model to use.
+            no_token (bool): Whether or not to require an API token. Set to False if you don't have an api key.
+
+        Returns:
+            HuggingFaceTextTranslationParser: The HuggingFaceTextTranslationParser object.
+
+        Usage:
+
+        1. Create a new model parser object with the model ID of the model to use.
+                parser = HuggingFaceTextTranslationParser("Helsinki-NLP/opus-mt-en-zh", use_api_token=False)
+        2. Add the model parser to the registry.
+                config.register_model_parser(parser)
+
+        If use_api_token is set to True, then the model parser will require an API token to be set in the environment variable HUGGING_FACE_API_TOKEN.
+
+
+        """
+        super().__init__()
+
+        token = None
+
+        if use_api_token:
+            # You are allowed to use Hugging Face for a bit before you get
+            # rate limited, in which case you will receive a clear error
+            token = get_api_key_from_environment(
+                "HUGGING_FACE_API_TOKEN", required=False
+            ).unwrap()
+
+        self.client = InferenceClient(model_id, token=token)
+
+    def id(self) -> str:
+        """
+        Returns an identifier for the Model Parser
+        """
+        return "HuggingFaceTextTranslationParser"
```

Comment:
Should we include "RemoteInference" in the parser IDs for remote inference models? I know we haven't figured out model vs model provider yet but in the meantime I think this would be helpful to distinguish in the UI

Reply:
Hmm, I'm not sure -- we already have a bit of a distinction with the 'Transformer' and 'Diffuser'. If we want to use RemoteInference would we also drop the 'Parser' for consistency with the local ones?

Reply:
I think that would be fine. Just felt like it would be easier to navigate if the naming were more explicit